### PR TITLE
Add sorted function to priority queue and move to utils

### PR DESCRIPTION
**Pull request by danield9tqh**
## Summary
Planning to re-use priority queue for other use cases besides just the mempool so moving it to a shared utils package. Also having a `sorted`function that copies the queue and generates the items in order is useful for both use cases so adding that as well

## Testing Plan
Unit tests

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish/src/memPool/memPool.ts [hunk 1]
```diff
@@ -171,22 +171,15 @@ export class MemPool {
...
+    for (const { hash } of this.feeRateQueue.sorted()) {
+      const transaction = this.transactions.get(hash)
 
       // The queue is cloned above, but this.transactions is not, so the
       // transaction may be removed from this.transactions while iterating.
-      if (transaction === undefined) {
-        continue
+      if (transaction !== undefined) {
+        yield transaction
       }
```
Comment (patnir) [inline ironfish/src/memPool/memPool.ts:182]: could you do `if (transaction)` ?

Comment (danield9tqh) [reply ironfish/src/memPool/memPool.ts:182]: yea I can change to that

**Review by patnir — APPROVED** [review]
Looks good I left a comment

File: ironfish/src/memPool/index.ts [hunk 2]
```diff
@@ -3,4 +3,4 @@
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 export * from './memPool'
 export * from './feeEstimator'
-export * from './priorityQueue'
+export { PriorityQueue } from '../utils/priorityQueue'
```
Comment (mat-if) [inline ironfish/src/memPool/index.ts:6]: Is it intentional that we're exporting this from a higher directory here down in the mempool index? This seems kind of strange

Comment (danield9tqh) [reply ironfish/src/memPool/index.ts:6]: Yea was just about to change this. Also added SDK change to next draft release

**Pull request merged**